### ISSUE TITLE
[bug] Sign error in JitTraceGraph_ELBO

### ISSUE THREAD

**Issue by jamestwebber** [pyro-ppl/pyro#2798]
Trying out this class and ended up with a rather strange looking loss curve (I was using a LR scheduler which explains the waviness):

![elbo_bug](https://user-images.githubusercontent.com/963647/113894124-b22ac200-9795-11eb-84ee-c2c0fab6453c.png)

It looks like the training losses have a sign error, although they are still optimizing in the right direction. This bug doesn't appear in `TraceGraph_ELBO` or `JitTrace_ELBO`.

Looking at the code, I think this is just because the JITed class overrides `loss_and_grads` and lacks a negative sign which appears in the non-JIT version. The validation curve is generated via `SVI.evalulate_loss` which calls `TraceGraph_ELBO.loss`  and includes the negation.

So it seems like a simple fix, but before I made a PR I wanted to flag the issue in case I'm missing something. There seem to be negations in a couple places and it seems easy to mess up

**Comment by fritzo** [comment]
Good catch @jamestwebber!  We had a similar bug #1982 and added what I thought was a thorough regression test in #1988.  I think the first step in a PR to fix your bug would be a similar regression test (say in `test_valid_models.py`) that fails for the current `JitTraceGraph_ELBO` implementation.

**Comment by jamestwebber** [comment]
Ah yes, digging into the existing tests (just running `test_plate_ok`) it looks like the loss ends up at 0.0 and so -0.0 passes just fine.

I'll write a PR in a bit.

**Comment by r3v1** [comment]
Facing the same issue with Trace_ELBO(), Beta distribution and MNIST data. Any advances on this?

**Comment by fritzo** [comment]
Hmm I guess @jamestwebber added failing tests in #2799, but we haven't gotten around to making all those pass. I can add this to my TODO list
